### PR TITLE
doc: test_and_optimize: misc edits

### PR DESCRIPTION
**Pull request by greg-fer**
Aligned steps about connecting to kits and emulators. Added link to Dev Academy course.
Removed mention of a missing step in the device hardware guides. NCSDK-22325.

### CODE REVIEW

**Review by alwa-nordic — APPROVED** [review]

**Review by trantanen — APPROVED** [review]
Modem shell looks good

**Review by MarkusLassila — APPROVED** [review]
With SLM, I think that it is fine to remove the Termite and Teraterm at this point.

File: applications/serial_lte_modem/doc/slm_description.rst [hunk 1]
```diff
@@ -399,23 +399,14 @@ To connect to an nRF91 Series DK with a PC:
    Using the Cellular Monitor app in combination with the nRF Connect Serial Terminal shows how the modem responds to the different modem commands.
    You can then use this connection to send or receive AT commands over UART, and to see the log output of the development kit.
 
-   Alternatively, you can use a terminal emulator like `Termite`_, `Teraterm`_, or PuTTY to establish a terminal connection to the development kit, using the :ref:`default serial port connection settings <test_and_optimize>`.
+   Alternatively, you can use a terminal emulator like PuTTY to establish a terminal connection to the development kit, using the :ref:`default serial port connection settings <test_and_optimize>`.
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/slm_description.rst:402]: Should this recommend nRF Connect Serial Terminal instead?

**Review by wiba-nordic — APPROVED** [review]
One question, but otherwise looks good to me.

**Review by annwoj — APPROVED** [review]

**Review by FrancescoSer — APPROVED** [review]

File: applications/serial_lte_modem/doc/slm_description.rst (hunk 1 above)
Comment (greg-fer) [reply applications/serial_lte_modem/doc/slm_description.rst:402]: Discussed on Teams and edited.

**Review by wiba-nordic — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**